### PR TITLE
Fixes transaction rollback to handle errors thrown while building the sql query

### PR DESCRIPTION
**Pull request by rcheman**

### CODE REVIEW

**Review by Solit95 — COMMENTED** [review]
I looks like on line 88 in Model.ts you remove the flip of the flag of transaction InProgress back to false but dont flip the flag anymore in the end Transaction function

**Comment by rcheman** [comment]
transactionInProgress flag now gets changed in Rollback function to false.

File: src/functions/errorMessages.ts [hunk 1]
```diff
@@ -9,18 +9,27 @@ class IncorrectData extends Error {
 export function checkQueryString(
   length: number,
   method: string,
-  model: string,
+  model: any,
```
Comment (alexanderlin) [inline src/functions/errorMessages.ts:12]: should the type be `Model` because we are now passing the actual model to the function?

Comment (rcheman) [reply src/functions/errorMessages.ts:12]: That is an excellent point. I think I had tried that before and thought it was causing problems, but it doesn't seem to be when I change it to type Model now. Thanks for catching that!!

**Review by alexanderlin — APPROVED** [review]

**Review by Solit95 — APPROVED** [review]

**Pull request merged**